src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotResolvedJavaMethodImpl.java

```diff
@@ -573,6 +573,19 @@ public boolean isDefault() {
         return ((getModifiers() & mask) == Modifier.PUBLIC) && getDeclaringClass().isInterface();
     }
 
+    /*
+     * Currently in hotspot an instance method can either be a "normal" or an "overpass" method.
+     * Overpass methods are instance methods which are created when otherwise a valid candidate
+     * for method resolution would not be found.
+     */
+    @Override
+    public boolean isDeclared() {
+        if (isConstructor() || isStatic()) {
```

Comment:
`isStatic()` -> `isClassInitializer()`

Reply:
Looks like you did not yet make the isClassInitializer fix. This also implies some missing test coverage in TestResolvedJavaType. Can you please address both these issues.